src/Cache/FileSystemCacheItemPool.php

```diff
@@ -0,0 +1,212 @@
+<?php
+/**
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace Google\Auth\Cache;
+
+use ErrorException;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\CacheItemInterface;
+
+class FileSystemCacheItemPool implements CacheItemPoolInterface
+{
+    private string $cachePath = 'cache/';
+    private array $deferredPool = [];
+
+    public function __construct($options = [])
+    {
+        if (array_key_exists('path', $options)) {
+            $this->cachePath = $options['path'];
+        }
+
+        if (is_dir($this->cachePath)) {
+            return true;
+        }
```

Comment:
Oh my. Yeah haha I basically want to handle the case where the cache folder already exists. But returning true from a `__construct` is definitely a boo boo haha. 

In that topic, what do we think it would be the best option when a cache folder already exists? Just re use it? Raise an exception? Clear it?